src/handlers/ResponseHandlers.ts

```diff
@@ -38,6 +39,42 @@ const error: ResponseHandler = {
     },
 };
 
+const blockedByWAFHandler: ResponseHandler = {
+    canProcess: (response) => response.headers.get('x-coveo-waf-action') === 'block',
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = 'Blocked for security reason'; // TODO: technical writer
+        platformError.detail = 'The web application firewall deemed that request to be dangerous so it was blocked'; // TODO: technical writer
+        platformError.status = response.status;
+        throw platformError;
+    },
+};
+
+const badGatewayHandler: ResponseHandler = {
+    canProcess: (response) => response.status === 502,
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = 'The endpoint is not reachable ...'; // TODO: technical writer
+        platformError.detail = 'Probably because bad gateway'; // TODO: technical writer
```

Comment:
Would that make sense? 

```suggestion
        platformError.detail = 'The service is currently unable to reach the necessary endpoint, likely due to a bad gateway. Please try your request again later.';
```